websites/D/Desu-Online/presence.ts

```diff
@@ -0,0 +1,157 @@
+const presence = new Presence({
+		clientId: "580032576434077707",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let video = {
+	current: 0,
+	duration: 0,
+	paused: true,
+};
+
+enum Assets {
+	Search = "https://i.imgur.com/1VwDv4D.png",
+	Home = "https://i.imgur.com/xVhTZcB.png",
+	Play = "https://i.imgur.com/Zd2WTR1.png",
+	Pause = "https://i.imgur.com/LV9xIjD.png",
+	Logo = "https://i.imgur.com/KEkxfKO.png",
+}
+
+presence.on(
+	"iFrameData",
+	(data: { current: number; duration: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname, search } = document.location;
+	presenceData.smallImageKey = Assets.Search;
```

Comment:
Move this to under the search if

Reply:
i mean, this:
```
presenceData.smallImageKey = Assets.Search;
if (x) {
   presenceData.details = x;
}
else if (y) {
   presenceData.details = y;
}
else if (z) {
   presenceData.details = z;
}
```
is better than this:
```
if (x) {
   presenceData.details = x;
   presenceData.smallImageKey = Assets.Search;
}
else if (y) {
   presenceData.details = y;
   presenceData.smallImageKey = Assets.Search;
}
else if (z) {
   presenceData.details = z;
   presenceData.smallImageKey = Assets.Search;
}
```

Reply:
and if pathname needs a different image, I just overwrite it like I did on the home page:
```ts
else if (pathname === "/") {
   presenceData.details = "Przegląda stronę główną...";
   presenceData.smallImageKey = Assets.Home;
}
```

Reply:
ok agreed